### PR TITLE
Improve Nutanix credential manifest creation

### PR DESCRIPTION
**Pull request by adiantum**
Change ipi-conf-nutanix-manual-creds step to use ccoctl instead of hardcoded manifest creation

### CODE REVIEW

**Comment by JoelSpeed** [comment]
I'd like to see this run before we merge, but otherwise, changes look good to me
/pj-rehearse pull-ci-openshift-cloud-provider-nutanix-main-e2e-nutanix

**Comment by adiantum** [comment]
/pj-rehearse pull-ci-openshift-cloud-provider-nutanix-main-e2e-nutanix

**Comment by adiantum** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-nutanix-ovn

File: ci-operator/step-registry/ipi/conf/nutanix/manual-creds/ipi-conf-nutanix-manual-creds-commands.sh [hunk 1]
```diff
@@ -6,17 +6,35 @@ set -o pipefail
...
+oc adm release extract --credentials-requests --cloud=nutanix --to "${CR_DIR}" "${RELEASE_IMAGE_LATEST}"
+
+# Create the credentials request manifest
+ccoctl nutanix create-shared-secrets \
+      --credentials-requests-dir="${CR_DIR}" \
+      --output_dir="/tmp" \
+      --credentials-source-filepath="${SHARED_DIR}/credentials"
+
+# Copy the manifest to the shared directory
+pushd /tmp
```
Comment (yanhua121) [inline ci-operator/step-registry/ipi/conf/nutanix/manual-creds/ipi-conf-nutanix-manual-creds-commands.sh:36]: I think the credentialRequest yaml files are in the dir /tmp/manifests.

File: ci-operator/step-registry/ipi/conf/nutanix/manual-creds/ipi-conf-nutanix-manual-creds-commands.sh [hunk 2]
```diff
@@ -6,17 +6,35 @@ set -o pipefail
...
+for FILE in *.yaml; do
+  cp "${FILE}" "${MANIFEST_PREFIX}_${FILE}"
```
Comment (yanhua121) [inline ci-operator/step-registry/ipi/conf/nutanix/manual-creds/ipi-conf-nutanix-manual-creds-commands.sh:38]: I think it should be:

cp "${FILE}" "${MANIFEST_PREFIX}/${FILE}"

Comment (adiantum) [reply ci-operator/step-registry/ipi/conf/nutanix/manual-creds/ipi-conf-nutanix-manual-creds-commands.sh:38]: No, it shouldn't.

Please take a look https://github.com/openshift/release/blob/master/ci-operator/step-registry/ipi/install/install/ipi-install-install-commands.sh#L364 that handle manifests before installation.

File: ci-operator/step-registry/ipi/conf/nutanix/manual-creds/ipi-conf-nutanix-manual-creds-commands.sh (hunk 1 above)
Comment (adiantum) [reply ci-operator/step-registry/ipi/conf/nutanix/manual-creds/ipi-conf-nutanix-manual-creds-commands.sh:36]: Good catch. Thank you!

**Comment by adiantum** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-nutanix-ovn

File: ci-operator/step-registry/ipi/conf/nutanix/manual-creds/ipi-conf-nutanix-manual-creds-commands.sh (hunk 2 above)
Comment (yanhua121) [reply ci-operator/step-registry/ipi/conf/nutanix/manual-creds/ipi-conf-nutanix-manual-creds-commands.sh:38]: Okay, got it.

**Comment by yanhua121** [comment]
@adiantum Could you squash the commit [Fix temp manifests directory](https://github.com/openshift/release/pull/35576/commits/590f8a70f27f149ff4efeb3bfa8674a6ef4bfd8d), for the cleaner commit history?

**Comment by adiantum** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-nutanix-ovn

File: ci-operator/step-registry/ipi/conf/nutanix/manual-creds/ipi-conf-nutanix-manual-creds-commands.sh [hunk 3]
```diff
@@ -6,17 +6,35 @@ set -o pipefail
...
 EOF
+
+# Extract credential requests
+oc registry login
+oc adm release extract --credentials-requests --cloud=nutanix --to "${CR_DIR}" "${RELEASE_IMAGE_LATEST}"
+
+# Create the credentials request manifest
+ccoctl nutanix create-shared-secrets \
+      --credentials-requests-dir="${CR_DIR}" \
+      --output_dir="/tmp" \
```
Comment (yanhua121) [inline ci-operator/step-registry/ipi/conf/nutanix/manual-creds/ipi-conf-nutanix-manual-creds-commands.sh:32]: The option is "--output-dir", not "--output_dir".

**Comment by adiantum** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-nutanix-ovn

**Comment by adiantum** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-nutanix-ovn

**Comment by JoelSpeed** [comment]
With the latest changes, the Machine API doesn't have credentials so something isn't working right here
/hold until we can see MAPI with correct credentials

**Comment by adiantum** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-nutanix-ovn

File: ci-operator/step-registry/ipi/conf/nutanix/manual-creds/ipi-conf-nutanix-manual-creds-commands.sh [hunk 4]
```diff
@@ -6,17 +6,42 @@ set -o pipefail
...
-metadata:
-   name: nutanix-credentials
-   namespace: openshift-machine-api
-type: Opaque
-data:
-  credentials: ${credentials}
+CR_DIR="/tmp/credentials_request"
+mkdir -p "${CR_DIR}"
+
+MANIFEST_PREFIX="${SHARED_DIR}/manifests"
```
Comment (yanhua121) [inline ci-operator/step-registry/ipi/conf/nutanix/manual-creds/ipi-conf-nutanix-manual-creds-commands.sh:12]: Please change "/manifests" to "/manifest", the line should be:
MANIFEST_PREFIX="${SHARED_DIR}/manifest"

Look at https://github.com/openshift/release/blob/master/ci-operator/step-registry/ipi/install/install/ipi-install-install-commands.sh#L364:
find "${SHARED_DIR}" \( -name "manifest_*.yml" -o -name "manifest_*.yaml" \)

**Comment by adiantum** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-nutanix-ovn

**Comment by adiantum** [comment]
/pj-rehearse pull-ci-openshift-installer-release-4.13-e2e-nutanix-ovn

**Comment by JoelSpeed** [comment]
/approve
/lgtm
/pj-rehearse ack

I can see the test didn't pass, but having checked, MAO got credentials and worked as expected, so I think this change is good

**Pull request merged**